lib/puppet/provider/user/directoryservice.rb

```diff
@@ -401,6 +401,13 @@ def iterations=(value)
   # we have to treat the ds cache just like you would in the password=
   # method.
   def salt=(value)
+    if (Puppet::Util::Package.versioncmp(self.class.get_os_version, '10.15') >= 0)
+      if value.length != 64
+        raise Puppet::Error, "macOS versions 10.15 and higher require the salt to be 32-bytes. Since Puppet's user 
+        resource requires the value to be hex encoded, the length of the salt's
+        string must be 64. Please check your salt and try again."
+      end
+    end
```

Comment:
Ah it looks like your string formatting changes on the VM didn't make it here. Could you join the string together so there aren't newlines?

I looked into the difference between calling `self.fail` and `raise Puppet::Error`. They are equivalent in cases like this where you just want to raise a Puppet::Error exception. The case where it's important to call [`self.fail defined here`](https://github.com/puppetlabs/puppet/blob/b5b96823e8e5874a64b86de1d93442abbdfb4eae/lib/puppet/util/errors.rb#L147) is if your provider has rescued an exception and you want to wrap it in a `Puppet::Error` while preserving the original exception's context, for example as done in https://github.com/puppetlabs/puppet/blob/b5b96823e8e5874a64b86de1d93442abbdfb4eae/lib/puppet/provider/package/puppetserver_gem.rb#L93

Reply:
Yes, sorry, I thought I had made the changes before I committed. Thank you!